lib/generator.js

```diff
@@ -573,38 +577,47 @@ class Generator {
         // We did our best. Proceed with installation...
       }
     }
-  
+
     const debugMessage = force ? logMessage.TEMPLATE_INSTALL_FLAG_MSG : logMessage.TEMPLATE_INSTALL_DISK_MSG;
     log.debug(logMessage.installationDebugMessage(debugMessage));
-  
+
     if (isFileSystemPath(this.templateName)) log.debug(logMessage.NPM_INSTALL_TRIGGER);
-  
-    const arbOptions = {
-      path: ROOT_DIR,
-    };
-    if (this.registry) {
+
+    const npmPath = process.execPath.replace('/bin/node','/lib/node_modules/npm/');
+
+    const config = new Config({
+      definitions,
+      flatten,
+      shorthands,
+      npmPath
+    });
+
+    await config.load();
+
+    const arbOptions = {...{path: ROOT_DIR}, ...config.flat};
+
+    if (Object.keys(this.registry).length !== 0) {
```

Comment:
why we need to change it like this?

also, am I correct assuming `registry` setting passes to a library will be ignored if config is found?

Reply:
During my local test the `if (this.registry)` was always true because of the default value of the constructor of Generator ( `registry = {}` ).
In this code the config from npm are loaded first in the arbOptions and if something is present in registry arbOptions will be loaded with it (like a manual overloading)

Reply:
oh, totally makes sense, thanks